File: modified_scripts/c++/matrix_multiplication/deepseek-coder_6.7b.cpp

vector<vector<int>> matrix_multiplication(vector<vector<int>>& A, vector<vector<int>>& B) {
    int n = A.size();
    vector<vector<int>> result(n, vector<int>(n));
    
    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
            for (int k = 0; k < n; k++) {
                result[i][j] += A[i][k] * B[k][j];
            }
        }
    }
    
    return result;
}

File: modified_scripts/rust/matrix_multiplication/deepseek-coder_6.7b.rs

pub fn matrix_multiplication(a: &Vec<Vec<f64>>, b: &Vec<Vec<f64>>) -> Vec<Vec<f64>> {
    let a_rows = a.len();
    let a_cols = a[0].len();
    let b_cols = b[0].len();
    
    let mut result: Vec<Vec<f64>> = vec![vec![(0.0; b_cols)]; a_rows];
    
    for i in 0..a_rows {
        for j in 0..b_cols {
            for k in 0..a_cols {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    
    result
}